core/src/navigation/services/navigation.js

```diff
@@ -384,10 +384,26 @@ class NavigationClass {
       });
       updatedCompData.selectedNode = selectedNode || lastElement;
       updatedCompData.children = groupedChildren;
+      const isExpandCategoriesByNavigation = LuigiConfig.getConfigValue('settings.expandCategoryByNavigation');
+      if (isExpandCategoriesByNavigation) {
+        this.expandCategoriesByNavigationFn(updatedCompData.children, updatedCompData.selectedNode);
+      }
     }
     return updatedCompData;
   }
 
+  expandCategoriesByNavigationFn(sortedChildrenEntries, selectedNode) {
```

Comment:
Looks good. Only ask if you could add a comment JSDOC on this function `expandCategoriesByNavigationFn`